Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java

```diff
@@ -193,6 +193,44 @@ public void setSelectedPhysicalNode(TreeNode selectedPhysicalNode) {
         return Optional.ofNullable(dataObject instanceof MediaUnit ? (MediaUnit) dataObject : null);
     }
 
+    /**
+     * Select given MediaUnit in physical structure tree.
+     *
+     * @param mediaUnit
+     *          MediaUnit to be selected in physical structure tree
+     */
+    public void selectMediaUnit(MediaUnit mediaUnit) {
+        TreeNode matchingTreeNode = getMatchingTreeNode(getPhysicalTree(), mediaUnit);
+        if (Objects.nonNull(matchingTreeNode)) {
+            updatePhysicalNodeSelection(matchingTreeNode);
+            matchingTreeNode.setSelected(true);
+        }
+    }
+
+    private TreeNode getMatchingTreeNode(TreeNode parent, MediaUnit mediaUnit) {
+        TreeNode matchingTreeNode = null;
+        for (TreeNode treeNode : parent.getChildren()) {
+            if (Objects.nonNull(treeNode.getData()) && treeNode.getData() instanceof StructureTreeNode) {
+                StructureTreeNode structureTreeNode = (StructureTreeNode) treeNode.getData();
+                if (Objects.nonNull(structureTreeNode.getDataObject()) && structureTreeNode.getDataObject() instanceof MediaUnit) {
+                    MediaUnit currentMediaUnit = (MediaUnit) structureTreeNode.getDataObject();
+                    if (Objects.nonNull(currentMediaUnit.getDivId())
```

Comment:
```
                    if (Objects.nonNull(currentMediaUnit.getDivId())
                            && currentMediaUnit.getDivId().equals(mediaUnit.getDivId())) {
```

equals method uses instanceof to check against null so no need to check for it here

Of course it is also not a mistake to do it here, but this method itself already has high cognitive complexity (mainly because of  2 breaks inside for loop) so this changes will decrease it at least a bit and the code will be slightly easier to read.